src/entries/popup/pages/settings/customRpc.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useCallback, useState } from 'react';
+
+import { useCustomRPCsStore } from '~/core/state/customRPC';
+import { isValidUrl } from '~/core/utils/connectedApps';
+import { Box, Button, Inline, Stack, Text } from '~/design-system';
+import { Input } from '~/design-system/components/Input/Input';
+
+import { maskInput } from '../../components/InputMask/utils';
+
+export function SettingsNetworksCustomRPC() {
+  const { customRPCs, addCustomRPC } = useCustomRPCsStore();
+  const [customRPC, setCustomRPC] = useState<{
+    rpcUrl?: string;
+    chainId?: number;
+    name?: string;
+    symbol?: string;
+    explorerUrl?: string;
```

Comment:
Should we add another field `explorerName` ? We could use it everywhere where we say stuff like "view on Etherscan"

Reply:
I think we're using the url hostname rn, (I kinda like that rly) but should change so that we grab from the rpc config instead of wagmi 
<img width="518" alt="Screenshot 2023-10-27 at 13 55 57" src="https://github.com/rainbow-me/browser-extension/assets/6232729/07b4dd54-e1c9-4f02-b593-7bd4766c9580">